src/main/java/org/quiltmc/asmr/verifier/Checker.java

```diff
@@ -170,6 +195,24 @@ private static boolean fieldBlacklisted(String clazz, String name, String descri
 		return BLACKLIST_FIELDS.contains(new VerificationDesc(realOwner, name, descriptor));
 	}
 
+	private static boolean typeAllowedForLambdaCapture(String name) throws ClassNotFoundException {
+		if ("VZCBSIFJD".contains(name)) {
+			return true;
+		}
+
+		if (WHITELIST_LAMBDA_CAPTURE_CLASSES.contains(name)) {
+			return true;
+		}
+
+		Class<?> cl = Class.forName(name.replace('/', '.'));
+		return cl.isAnnotationPresent(AllowLambdaCapture.class);
+	}
+
+	private static boolean classAllowedForLambdaReturn(String name) throws ClassNotFoundException {
+		Class<?> cl = Class.forName(name.replace('/', '.'));
+		return !cl.isAnnotationPresent(ApiStatus.NonExtendable.class);
+	}
```

Comment:
No, this is not how we're doing this. We detect the presence of annotations via an annotation processor.

Reply:
Added annotation processors, probably still doing something wrong though.